kubemarine/procedures/restore.py

```diff
@@ -135,19 +136,25 @@ def restore_thirdparties(cluster):
 
     install.system_prepare_thirdparties(cluster)
 
-
+    
 def import_nodes(cluster):
     for node in cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True):
-        node['connection'].put(os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name']),
-                               '/tmp/kubemarine-backup.tar.gz')
-        cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
-
-    cluster.log.debug('Unpacking backup...')
-    result = cluster.nodes['all'].sudo(
-        'chattr -i /etc/resolv.conf; sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite && sudo chattr +i /etc/resolv.conf')
-    cluster.log.debug(result)
-
-
+        node_backup_file = os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name'])
+        if os.path.exists(node_backup_file):
+            node['connection'].put(node_backup_file, '/tmp/kubemarine-backup.tar.gz')
+            cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
+
+            if not cluster.procedure_inventory.get('backup_plan', {}).get('nodes', {}).get('resolv.conf'):
+                restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz --exclude=/etc/resolv.conf -C / --overwrite'
+            else:
+                restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite'
+
+            result = node['connection'].sudo(restore_command)
```

Comment:
Previously we used nodes['all'] to unpack the archives on all nodes in parallel. We can still support parallel unpacking even in this case, but it is slightly more challenging. Consider using of `kubemarine.core.executor.RemoteExecutor`.
* It might also be possible to even upload the archive on nodes in parallel using the same RemoteExecutor instance.

Reply:
* get_ordered_members_list method is called exactly once before and after these changes, so the number of calls did not change.
* Using nodes['all'] to unpack archives on all nodes in each loop iteration is incorrect.
* RemoteExecutor is surely working feature, as it is very often used in Kubemarine in practice.

Reply:
Example usage https://github.com/Netcracker/KubeMarine/blob/0.15.1/kubemarine/haproxy.py#L112